src/Components/CardHorizontal/CardHorizontal.jsx

```diff
@@ -1,24 +1,31 @@
-import { useData } from "../../Helper";
+import {
+  useAuth,
+  removeFromCart,
+  addToWishlist,
+  changeItemQuantity,
+} from "../../Helper";
 import "./CardHorizontal.css";
 
 const CardHorizontal = ({ product }) => {
-  const { inWishlist, dispatch } = useData();
+  const { userState, dispatchUserState, inWishlist } = useAuth();
 
-  const itemQuantityHandler = (e, product) => {
-    dispatch({
-      type: "CHANGE_ITEM_QUANTITY",
-      payload: { id: product.id, qty: e.target.value },
-    });
+  const itemQuantityHandler = (e) => {
+    changeItemQuantity(
+      product,
+      userState.token,
+      dispatchUserState,
+      e.target.value
+    );
   };
 
-  const removeCartItemHandler = (_id) => {
-    dispatch({ type: "REMOVE_FROM_CART", payload: _id });
+  const removeCartItemHandler = () => {
+    removeFromCart(product, userState.token, dispatchUserState);
```

Comment:
method only needs `_id`
```suggestion
    removeFromCart(product._id, userState.token, dispatchUserState);
```

Reply:
fixed in https://github.com/saadahmadhere/elmart/pull/13/commits/1b5c9c8caec12d29427cb645b878d98bdb564977